File: src/main/GO/leetcode/lists/offer/Offer32_1.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2023/1/3 19:00
 */

func levelOrder32_1(root *TreeNode) []int {

	if root == nil {
		return []int{}
	}
	q := make([]*TreeNode, 0)
	q = append(q, root)
	list := make([]int, 0)
	for len(q) > 0 {
		node := q[0]
		q = q[1:]
		list = append(list, node.Val)
		if node.Left != nil {
			q = append(q, node.Left)
		}
		if node.Right != nil {
			q = append(q, node.Right)
		}
	}
	return list
}

File: src/main/java/leetcode/lists/offer/Offer32_1.java
package leetcode.lists.offer;

import common.*;

import java.util.*;

/**
 * @author jingxinwu
 * @date 2021-11-21 8:57 下午
 */
public class Offer32_1 {

    public int[] levelOrder(TreeNode root) {

        List<Integer> list = new ArrayList<>();

        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()) {

            TreeNode node = queue.poll();
            list.add(node.val);
            if (node.left != null) {
                queue.add(node.left);
            }
            if (node.right != null) {
                queue.add(node.right);
            }

        }
        int[] res = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            res[i] = list.get(i);
        }
        return res;
    }


}
